File: bits/136.single-number.py
#
# @lc app=leetcode id=136 lang=python3
#
# [136] Single Number
#

# @lc code=start
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        res = 0
        for sub in nums:
            res ^= sub
        return res
            
# @lc code=end

File: bit/136.single-number.cpp
/*
 * @lc app=leetcode id=136 lang=cpp
 *
 * [136] Single Number
 */

#include "bits/stdc++.h"

using namespace std;

// @lc code=start
class Solution {
public:
  int singleNumber(vector<int> &nums) {
    int once=0;
    for (const auto &val : nums) {
      once ^= val;
    }
    return once;
  }
};
// @lc code=end
